### ISSUE TITLE
Can I specify the size of C type long?

### ISSUE THREAD

**Issue by zyz9740** [emscripten-core/emscripten#17009]
Code:
```
#include <stdio.h>

int main() {
    // C: 8, wasm: 4
    printf("%d\n", sizeof(long));
    // C: 8, wasm: 8
    printf("%d\n", sizeof(long long));
    // C: 0, wasm: 1
    printf("%d\n", -1 > 0x8D86D88CL);
    return 0;
}
```
When I compile this C program to wasm and run it, it prints different value from C program. I want to know is it a intentional action or I can use some compile option to get a 8-bit type LONG ?

emcc -v:
 emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.10 (c3fe57af0504fe24fc4ba697feb8c204f3c80022)
clang version 15.0.0 (https://github.com/llvm/llvm-project 8bc29d14273b05b05d5a56e34c07948dc2c770d3)
Target: wasm32-unknown-emscripten
Thread model: posix
InstalledDir: /home/oem/wasm/emsdk/upstream/bin

**Comment by tlively** [comment]
No, I don't think this is something that can be changed. There are a bunch of platforms out there with 32-bit longs, so hopefully most portable code should be able to handle that.

**Comment by sbc100** [comment]
Indeed, if you run that same code through `clang -m32` or `gcc -m32` on an x86 machine you will see the same result as with wasm (4, 8 ,1).    This is norm for and how all 32-bit (ILP32) targets works.   wasm is just one of them. See https://docs.oracle.com/cd/E19620-01/805-3024/lp64-1/index.html

**Comment by sgbeal** [comment]
Don't forget that C99 defines `<stdint.h>`, which specifies fixed-sized integer types like `int64_t` and `int32_t`. The portable printf/scanf specifiers for them are (for whatever reason) available separately in `<inttypes.h>`. Details and example:

<https://en.cppreference.com/w/c/types/integer>
